spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/discovery/KubernetesAwareServiceInstance.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.commons.discovery;
+
+import java.net.URI;
+import java.util.Map;
+
+import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.HTTP;
+import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.HTTPS;
+import static org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryConstants.NAMESPACE_METADATA_KEY;
+
+/**
+ * @author wind57
+ * @param instanceId the id of the instance.
+ * @param serviceId the id of the service.
+ * @param host the address where the service instance can be found.
+ * @param port the port on which the service is running.
+ * @param metadata a map containing metadata.
+ * @param secure indicates whether the connection needs to be secure.
+ * @param namespace the namespace of the service.
+ * @param cluster the cluster the service resides in.
+ */
+public record KubernetesAwareServiceInstance(String instanceId, String serviceId, String host, int port,
```

Comment:
I would prefer we call this `DefaultKubernetesServiceInstance`.  That follows the naming convention we use for the most part.

Reply:
I can't tell you how much time I've spent thinking about proper names here. very much appreciated.